modules/core/src/main/java/org/apache/ignite/internal/event/EventListener.java

```diff
@@ -33,4 +36,19 @@ public interface EventListener<P extends EventParameters> {
      *         future will never be completed with {@code null} value.
      */
     CompletableFuture<Boolean> notify(P parameters, @Nullable Throwable exception);
+
+    /**
+     * Creates an adapter for a given callback.
+     *
+     * <p>Created listener will never return a future completed with {@code true}.
+     */
+    static <P extends EventParameters> EventListener<P> fromConsumer(Consumer<P> callback) {
```

Comment:
I don't think we should return a failed future if `exception != null`, because the listener contract doesn't state anything about that, but I agree with wrapping in `try-catch`

Reply:
Added try-catch